shell/components/form/PodAffinity.vue

```diff
@@ -98,8 +109,13 @@ export default {
     },
 
     allNamespaces() {
-      const inStore = this.$store.getters['currentStore'](NAMESPACE);
-      const choices = this.$store.getters[`${ inStore }/all`](NAMESPACE);
+      const inStore = this.overrideStore || this.$store.getters['currentStore'](NAMESPACE);
+      let choices = this.$store.getters[`${ inStore }/all`](NAMESPACE);
+
+      if (this.overrideStore === HARVESTER) {
```

Comment:
We are trying to reduce harvester-specific code in generic components as part of making a Harvester plugin. It is important to look for alternative solutions to this type of logic. See my comment in the harvester machine-config.

Reply:
I am going to remove the `overrideStore` prop, we can pass `namespaces` into podAffinity.